spec/services/task_service/gpt_generate_tests_spec.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe TaskService::GptGenerateTests do
+  describe '.new' do
+    subject(:gpt_generate_tests_service) { described_class.new(task:) }
+
+    let(:programming_language) { build(:programming_language, language: 'Ruby') }
+    let(:task) { build(:task, description: 'Sample Task', programming_language:) }
+
+    it 'assigns task' do
+      expect(gpt_generate_tests_service.instance_variable_get(:@task)).to be task
+    end
+
+    it 'assigns message from task description' do
+      expect(gpt_generate_tests_service.instance_variable_get(:@message)).to eq('Sample Task')
+    end
+
+    it 'assigns language from programming language' do
+      expect(gpt_generate_tests_service.instance_variable_get(:@language)).to eq('Ruby')
+    end
+
+    context 'when language is missing' do
+      let(:task) { build(:task, description: 'Sample Task', programming_language: nil) }
+
+      it 'raises MissingLanguageError' do
+        expect { gpt_generate_tests_service }.to raise_error(MissingLanguageError)
+      end
+    end
+  end
+
+  describe '#execute' do
+    let(:programming_language) { create(:programming_language, language: 'Python') }
+    let(:task) { create(:task, description: 'Create a Python script.', programming_language:) }
+    let(:gpt_generate_tests_service) { described_class.new(task:) }
+    let(:mock_client) { instance_double(OpenAI::Client) }
+
+    before do
+      allow(OpenAI::Client).to receive(:new).and_return(mock_client)
+      allow(mock_client).to receive(:chat).and_return('choices' => [{'message' => {'content' => "```Python\ndef test_script():\n  assert true```"}}])
+    end
+
+    it 'creates a test file related to the task' do
+      expect { gpt_generate_tests_service.execute }.to change(TaskFile, :count).by(1)
+      test_file = TaskFile.last
+      expect(test_file).to have_attributes(
+        content: "def test_script():\n  assert true",
+        name: 'test.py'
+      )
+    end
+
+    it 'creates a test instance related to the task' do
+      expect { gpt_generate_tests_service.execute }.to change(Test, :count).by(1)
+      test = Test.last
+      expect(test).to have_attributes(title: I18n.t('tests.model.generated_test'))
```

Comment:
You could directly access the tests of the task like this:
`task.reload.tests.last` the `reload` reloads the object from the database so you can see your changes.
With this you could also omit the `include` check.

This can also be applied to the example above.

Reply:
for `test_file = TaskFile.last` what should be the alternative in this case?

Reply:
This is the alternative used: `test_file = task.reload.tests.last.files.first`